### DISCUSSION TITLE
Day Planning Feature

### DISCUSSION THREAD

**Discussion by Glyphack** [frustak/koal#9] [Ideas]
Problem
Want to have a way to plan my day trough koal.

Planing is a series of steps you do before starting the day
to have your todo list and manage your time.

This is an example of a planning routine
http://glyphack.com/blog/planning-my-day/

Boundaries
We can start off with a very minimal planning feature that
will not force any routine to you, and customize it later as
we get some feedback from using it.
For the start it can have:

1. Help you go through your goals, review them, and add todos
2. Have a way to mark one of the Goal as todays focus for that day
3. Projects can have a repetitive todo which is added to todos in planning
4. Let user set a focus time. We only show this time and not remind etc.

Focus time: Helps to put aside some time for todos that require uninterrupted focus in that day.
For example if you want to write something down you need an hour to think and write down.
But a todo like follow up a PR review or merging does not require focus.
So we can plan when to do different tasks to save the days focus time on important stuff.

![image](https://user-images.githubusercontent.com/20788334/202910587-e14083c8-613c-4152-a44f-737c6739c83f.png)

**Comment by Glyphack** [comment]
# Server API Specs

APIs needed to complete the tasks are:
- Add todo (Already Available `createTodo`)
- Mark a goal as important 🔨 
- Set focus time and get 🔨 
- Inbox API: get undone todos and focus time 🔨

Challenges:
Focused Goal and focus time are only for a day, and should be reset the next day.
To address this, we can add new entities:
- FocusTime: this keeps track of the focus time in each day we query it and get the current day focus time.
- FocusGoal: this keeps track of what is the focus goal in each day and by querying the day we get the current focus goal.

**Comment by Glyphack** [comment]
https://github.com/frustak/koal/pull/10

**Comment by frustak** [comment]
Only the frontend remains?

**Reply by Glyphack** [reply]
Yes

**Comment by frustak** [comment]
#13 needs to be implemented first
